stl/inc/valarray

```diff
@@ -549,11 +532,6 @@ private:
         }
     }
 
-    void _Tidy_init() noexcept {

```

Comment:
I'm uncomfortable with the magnitude of this PR (+616 -446 lines) combined with its grab-bag nature of `noexcept` strengthening and data member initializer refactoring. Can you please separate these changes? Isolated `noexcept` and DMI changes are easier to review and less risky.